sqlobject/src/main/java/org/jdbi/v3/sqlobject/locator/SqlLocator.java

```diff
@@ -29,6 +29,21 @@ public interface SqlLocator {
      * @param method the method
      * @param config the config registry
      * @return the SQL for the given method.
+     * @deprecated use {@link #locate(Class, Object, Method, ConfigRegistry)}
      */
+    @Deprecated
     String locate(Class<?> sqlObjectType, Method method, ConfigRegistry config);
+
+    /**
+     * Locates and returns the SQL for the given SQL Object type, target and method.
+     *
+     * @param sqlObjectType the SQL object type
+     * @param target the target
+     * @param method the method
+     * @param config the config registry
+     * @return the SQL for the given method.
+     */
+    default String locate(Class<?> sqlObjectType, Object target, Method method, ConfigRegistry config) {
```

Comment:
the sqlObjectType and the method are related. The target should be either in front (if it is more important than the pair) or before config (if it is less important) but not split them.